### PR TITLE
add initial version of PSA compliance test documents

### PR DESCRIPTION
**Pull request by NirSonnenschein**
PSA compliance test documents

### CODE REVIEW

**Comment by NirSonnenschein** [comment]
documents the addition of test from this PR: https://github.com/ARMmbed/mbed-os/pull/9312

**Comment by AnotherButler** [comment]
How does this fit into PR #967 ? Our typical porting guide template includes testing information at the bottom of the porting guide. Does this include testing information for all of PSA instead of just attestation? Is that it?

**Comment by AnotherButler** [comment]
We require engineering review for all new/changed content, so I'm tagging @avolinski for content review.
If he's busy or someone else would be a better fit for this review, please tag a different engineer (who didn't write the PR).

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [inline docs/porting/psa/testing_psa_service_compliance.md]: It is not immediately clear to me how https://github.com/ARMmbed/mbed-os-psa-compliance-tests-example serves as an example of how to run the tests with a different configuration.

**Comment by AnotherButler** [comment]
@avolinski @alzix I talked with @NirSonnenschein in our docs reps meeting this morning, and I think this is in the right place. However, our [normal porting template](http://docs-previewer.herokuapp.com/documentation_docs/master/writing/mbed-os-vendor-partner-contributing-api-reference.html) usually contains more information. Does either of you know if there's a nontesting porting element to PSA compliance? If so, who will be providing this content?

**Comment by alekshex** [comment]
Hi @AnotherButler ,
psa compliance will have in addition an external example repository mainly for OOB.
later when mbedOS will be able to support multiple configurations per build, this repository will become obsolete
https://github.com/ARMmbed/mbed-os-psa-compliance-tests-example

i am not sure if we want to mention it in the documentation
i am not able to access the porting template link in your message it result in 'not found' page
http://docs-previewer.herokuapp.com/documentation_docs/master/writing/mbed-os-vendor-partner-contributing-api-reference.html

**Comment by AnotherButler** [comment]
I should have mentioned that link only works on the Arm network.

Do you and @ndevillard think a porting guide, if relevant, should wait until Mbed OS can support multiple configurations?

**Comment by alekshex** [comment]
Hi @AnotherButler ,
i am on the Arm network, still...
Not Found
The requested resource was not found on this server.

**Comment by AnotherButler** [comment]
Interesting. That works for me. I'll have to look into it. Here's the GH version: https://github.com/ARMmbed/documentation_docs/blob/master/Docs/Templates/mbed-os-port-template.md

**Comment by AnotherButler** [comment]
@avolinski or @orenc17 Could one of you please approve this PR if it's ready to merge? We require approval from an engineer who didn't create the PR before we can merge content changes/additions. (Or feel free to tag a different engineer for approval if you're not the right person.)

**Comment by GuyWi** [comment]
@avolinski  or @orenc17  , please review and approve this PR.
Thank you!

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md:7]: ### Tests directory layout

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: `ARM`,  `ARMC6`, `GCC_ARM` or  `IAR`

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: `<target>` is  your target platform

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: ... -n components-target_psa-tests-compliance_......
without mbed-os- prefix

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md:39]: Wildcards can be used to compile and run the entire set of PSA compliance tests:
`mbed test -t <toolchain> -m <target> -n components-target_psa-tests-compliance_*`

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: (Please note, the tests can run only on PSA enabled target platforms)

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: You can override the default Mbed OS cryptography configuration using a `mbed_app.json` file.
Or supplying your custom configuration file directly as command line compilation flag.

**Review by alekshex — CHANGES_REQUESTED** [review]

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md:39]: @avolinski which wildcards can you use? Or do you always use an asterisk (*)?

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: @avolinski, do we have a list of PSA-compliant platforms?

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: @avolinski , can we specify what the flag is?

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Fixed.

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Fixed.

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Modified.

Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Modified.

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [reply docs/porting/psa/testing_psa_service_compliance.md]: Or supplying your custom configuration file directly as command line parameter flag MBEDTLS_USER_CONFIG_FILE=<custom config file path>
By passing it to the mbed compilation command

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (NirSonnenschein) [reply docs/porting/psa/testing_psa_service_compliance.md:39]: * is the only relevant wildcard we are aware of in this case.

Comment (NirSonnenschein) [reply docs/porting/psa/testing_psa_service_compliance.md:39]: "*" github renders it as a bullet

**Comment by NirSonnenschein** [comment]
@GuyWi thanks the latest exit looks fine to me.
@avolinski thanks for the notes. can you approve the fixes by Guy or do you want anything else?

**Review by AnotherButler — APPROVED** [review]

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Updated.

**Comment by GuyWi** [comment]
Thanks, @NirSonnenschein !
@avolinski, please approve.

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md:55]: When the tests run, they output state information to the serial output. Below is an example of a successful run:

mbedgt: mbed-host-test-runner: started
mbedgt: checking for GCOV data...
mbedgt: test on hardware with target id: 190000006c0f1507036c0f1500000000000000002e127069
mbedgt: test suite 'components-target_psa-tests-compliance_its-test_s009' ............................ OK in 55.01 sec
        test case: 'Check1' .......................................................................... OK in 0.40 sec
mbedgt: mbed-host-test-runner: started
mbedgt: checking for GCOV data...
mbedgt: test on hardware with target id: 190000006c0f1507036c0f1500000000000000002e127069
mbedgt: test suite 'components-target_psa-tests-compliance_its-test_s008' ............................ OK in 55.21 sec
        test case: 'Check1' .......................................................................... OK in 0.10 sec
        test case: 'Check2' .......................................................................... OK in 0.09 sec

mbedgt: test suite report:
| target                          | platform_name       | test suite                                           | result  | elapsed_time (sec) | copy_method |
|---------------------------------|---------------------|------------------------------------------------------|---------|--------------------|-------------|
| CY8CKIT_062_WIFI_BT_PSA-GCC_ARM | CY8CKIT_062_WIFI_BT | components-target_psa-tests-compliance_its-test_s001 | OK      | 19.45              | default     |
| CY8CKIT_062_WIFI_BT_PSA-GCC_ARM | CY8CKIT_062_WIFI_BT | components-target_psa-tests-compliance_its-test_s002 | OK      | 55.34              | default     |

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: please remove this section and below and replace with one above. Thank you!

**Review by alekshex — CHANGES_REQUESTED** [review]

File: docs/porting/psa/testing_psa_service_compliance.md
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: @avolinski, updated.
Please review and approve.
Thanks!

**Review by alekshex — APPROVED** [review]

**Pull request merged**
